src/shared/ui/components/widget-tab/widget-tab.component.tsx

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useMemo } from 'react';
+
+import { useWidgetTabs } from '../../providers/widget-tabs';
+import { WidgetBase } from '../widget-base';
+import { classes } from '../../utils/classes';
+
+import { TabHandle } from './tab-handle.component';
+import { WidgetTabProperties } from './widget-tab.types';
+
+export const WidgetTab = ({
+  top,
+  theme,
+  twitterHandle,
+  tabImage,
+  tabName,
+  children,
+  className,
+  closeButtonClassName,
+  onClose,
+}: WidgetTabProperties) => {
+  const {
+    tabs,
+    preferredTab,
+    addWidgetTab,
+    setUserPreferredTab,
+    removeAllUserWidgets,
+  } = useWidgetTabs();
+
+  const userTabs = useMemo(() => {
+    return tabs[twitterHandle];
+  }, [tabs, twitterHandle]);
+
+  const userPreferredTab = preferredTab[twitterHandle] ?? userTabs?.at(0);
+
+  const left = useMemo(() => {
+    const index = userTabs?.indexOf(tabName) ?? -1;
+    if (index !== -1) {
+      return index * 100;
+    }
+
+    return 0;
+  }, [tabName, userTabs]);
+
+  useEffect(() => {
+    addWidgetTab(twitterHandle, tabName);
+  }, [addWidgetTab, tabName, twitterHandle]);
+
+  useEffect(() => {
+    if (userTabs === null && onClose) {
+      onClose();
+    }
+  }, [onClose, userTabs]);
```

Comment:
I am wondering if these useEffect could be removed and this could be handled by doing:
```tsx
      onClose={() => {
        removeAllUserWidgets(twitterHandle);
        onClose?.();
      }}
```

WDYT?

Reply:
That is a hacky way to close all widgets. Clicking on the `x` calls `removeAllUserWidgets`, which sets `userTabs` to `null`. This triggers a `useEffect`, causing other widgets to call their own `onClose` functions.